tessera-core/src/test/java/com/quorum/tessera/privacygroup/PrivacyGroupManagerTest.java

```diff
@@ -0,0 +1,449 @@
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.*;
+
+public class PrivacyGroupManagerTest {
+
+    private Enclave enclave;
+
+    private PrivacyGroupManager privacyGroupManager;
+
+    private PrivacyGroupDAO privacyGroupDAO;
+
+    private PrivacyGroupUtil privacyGroupUtil;
+
+    private BatchPrivacyGroupPublisher publisher;
+
+    private final PublicKey localKey = PublicKey.from("ownKey".getBytes());
+
+    @Before
+    public void setUp() {
+        enclave = mock(Enclave.class);
+        privacyGroupDAO = mock(PrivacyGroupDAO.class);
+        publisher = mock(BatchPrivacyGroupPublisher.class);
+        privacyGroupUtil = mock(PrivacyGroupUtil.class);
+        privacyGroupManager = new PrivacyGroupManagerImpl(enclave, privacyGroupDAO, publisher, privacyGroupUtil);
+
+        when(enclave.getPublicKeys()).thenReturn(Set.of(localKey));
+    }
+
+    @After
+    public void tearDown() {
+        verifyNoMoreInteractions(privacyGroupDAO, publisher);
+    }
+
+    @Test
+    public void testCreatePrivacyGroup() {
+
+        when(privacyGroupUtil.generateId(anyList(), any(byte[].class))).thenReturn("generatedId".getBytes());
+        when(privacyGroupUtil.generateLookupId(anyList())).thenReturn("lookup".getBytes());
+        when(privacyGroupUtil.encode(any())).thenReturn("encoded".getBytes());
+
+        final List<PublicKey> members = List.of(localKey, mock(PublicKey.class), mock(PublicKey.class));
+
+        doAnswer(
+            invocation -> {
+                Callable callable = invocation.getArgument(1);
+                callable.call();
+                return mock(PrivacyGroupEntity.class);
+            })
+            .when(privacyGroupDAO)
+            .save(any(), any());
+
+        final PrivacyGroup privacyGroup =
+            privacyGroupManager.createPrivacyGroup("name", "description", localKey, members, new byte[1]);
+
+        // Verify entity being saved has the correct values
+        ArgumentCaptor<PrivacyGroupEntity> argCaptor = ArgumentCaptor.forClass(PrivacyGroupEntity.class);
+        verify(privacyGroupDAO).save(argCaptor.capture(), any());
+        PrivacyGroupEntity savedEntity = argCaptor.getValue();
+        assertThat(savedEntity).isNotNull();
+        assertThat(savedEntity.getId()).isEqualTo("generatedId".getBytes());
+        assertThat(savedEntity.getLookupId()).isEqualTo("lookup".getBytes());
+        assertThat(savedEntity.getData()).isEqualTo("encoded".getBytes());
+
+        // Verify payload being distributed has the correct values
+        ArgumentCaptor<byte[]> payloadCaptor = ArgumentCaptor.forClass(byte[].class);
+        ArgumentCaptor<List<PublicKey>> recipientsCaptor = ArgumentCaptor.forClass(List.class);
+        verify(publisher).publishPrivacyGroup(payloadCaptor.capture(), recipientsCaptor.capture());
+        assertThat(payloadCaptor.getValue()).isEqualTo("encoded".getBytes());
+
+        List<PublicKey> forwardingMembers =
```

Comment:
non-blocking: as this is the test, shouldn't be this expected value hard-code instead of being generated by the same logic that exists in the code?

Reply:
Good point - thanks! Test updated